pkg/provisioners/gcppubsub/util/names.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+
+	eventduck "github.com/knative/eventing/pkg/apis/duck/v1alpha1"
+)
+
+// Underscore is used as a separator between the namespace and name so the concatenated version
+// can't accidentally overlap.
+
+// GenerateTopicName generates the GCP PubSub Topic name given the Knative Channel's namespace and
+// name.
+// Note that it must be stable. The controller and dispatcher rely on this being generated
+// identically.
+func GenerateTopicName(channelNamespace, channelName string) string {
+	return fmt.Sprintf("knative-eventing-channel_%s_%s", channelNamespace, channelName)
```

Comment:
is that `knative-eventing-channel` something that's needed for GCP PUB/SUB? or would others (e.g. Kafka) also benefit from that - to make the "topic names" more robust/stable ?

Reply:
I added the `knative-eventing-channel` prefix because these names will be visible in GCP (e.g. when you list all your PubSub Topics) and want people to be able to understand where they are in use. Ideally, we would include something to determine which K8s cluster you are on too, but I didn't see anything handy, so punted on that until later.

If Kafka is also visible from outside the K8s cluster, then I think the prefix is useful.